index.html

```diff
@@ -2907,31 +2853,47 @@ <h2>Navigation</h2>
   is <a>no longer open</a>, return <a>success</a> with
   data <a><code>null</code></a>.
 
- <li><p>Start a <var>timer</var>. If this algorithm has not
-  completed before <var>timer</var> reaches
-  the <a>session</a>&apos;s <a>session page load timeout</a> in
-  milliseconds, return an <a>error</a> with <a>error
-  code</a> <a>timeout</a>.
+ <li><p>Let <var>timeout</var> be <a>session
+ timeouts</a>&apos; [=timeouts/page load timeout=].
+
+ <li><p>If <var>timer</var> is null:
+
+  <ol>
+   <li><p>Set <var>timer</var> to a new <a>timer</a>.
+
+   <li><p>If <var>timeout</var> is not null:
+
+   <ol>
+
+    <li><p><a>Start the timer</a> with <var>timer</var>
+    and <var>timeout</var>.
+
+   </ol>
+
+  </ol>
+
+ <li><p>Run these steps, but [=abort when=] <var>timer</var>&apos;s
+   [=timer/timeout fired flag=] is set:
+
+ <ol>
 
- <li><p>If there is an ongoing attempt to <a>navigate</a>
+  <li><p>If there is an ongoing attempt to <a>navigate</a>
   <var>session</var>&apos;s <a>current browsing context</a> that has not
   yet <a>matured</a>, wait for navigation to <a>mature</a>.
 
- <li><p>Let <var>readiness target</var> be the <a>document
+  <li><p>Let <var>readiness target</var> be the <a>document
   readiness</a> state associated with the <var>session</var>&apos;s <a>page
   loading strategy</a>, which can be found in the <a>table of page
   load strategies</a>.
 
- <li><p>Wait for <var>session</var>&apos;s <a>current browsing context</a>&apos;s
-  <a>document readiness</a> state to reach
-  <var>readiness target</var>, or for the
-  <a>session page load timeout</a> to pass, whichever
-  occurs sooner.
+  <li><p>Wait for <var>session</var>&apos;s <a>current browsing
+  context</a>&apos;s <a>document readiness</a> state to reach
+  <var>readiness target</var>.
+
+ </ol>
 
- <li><p>If the previous step completed by the
-  <a>session page load timeout</a> being reached
-  and the browser does not have an active <a>user prompt</a>,
-  return <a>error</a> with <a>error code</a> <a>timeout</a>.
```

Comment:
We are still failing with a `timeout`. It happens in the case of an `abort`. See line 2875 for that extra check. Does that solve your question?

Reply:
Right, my question is more that before this PR, if a user prompt was active when the timeout was reached, these steps would have returned "success".  But now an error is returned. It seems like that makes it impossible for tests like the above WPT test to pass.

Reply:
The above referenced test succeeds because the navigation steps for classic will return success when any user prompt is opened. As such it will not run into a timeout.

For Webdriver BiDi we cannot return immediately, and there is still https://github.com/w3c/webdriver-bidi/issues/556 to get fixed so that prompts can be automatically handled in case of no subscription for the `browsingContext.userPromptOpened` event is present. With such an event subscribed it will still succeed. See [this BiDi wpt test](https://github.com/web-platform-tests/wpt/blob/8be9ad55c941b65de8bc725fdc72eabca737ba43/webdriver/tests/bidi/browsing_context/user_prompt_opened/user_prompt_opened.py#L38-L65).

Reply:
> The above referenced test succeeds because the navigation steps for classic will return success when any user prompt is opened. As such it will not run into a timeout.

I must be missing something here because I don't see where the navigation steps return success upon a user prompt being opened. The [Navigate To](https://w3c.github.io/webdriver/#dfn-navigate-to) endpoint handles users prompts before navigation, but I don't see where it checks for prompts once the navigation begins?

Reply:
Sorry for the delay in response. We discussed it yesterday and yes, we actually lost this part. I filed https://github.com/w3c/webdriver/issues/1857 to bring it back and maybe have further improvements.

Thanks again for bringing it up!